web/src/components/FilteredConnection.tsx

```diff
@@ -500,15 +569,32 @@ export class FilteredConnection<N, NP = {}, C extends Connection<N> = Connection
                         if (this.props.onUpdate) {
                             this.props.onUpdate(connectionOrError)
                         }
+
+                        // If our response has a page info object, try to pull the end cursor out of
+                        // it so we can pass it to the subsequent request. We'll either update or
+                        // unset the after cursor here so it is not used twice.
+                        if (connectionOrError && !isErrorLike(connectionOrError) && connectionOrError.pageInfo) {
+                            after = connectionOrError.pageInfo.endCursor || undefined
+                        } else {
+                            resetCursor()
+                        }
                     })
                 )
                 .subscribe(stateUpdate => this.setState(stateUpdate), err => console.error(err))
         )
 
         this.subscriptions.add(
             this.showMoreClicks
-                .pipe(map(() => this.state.first * 2))
-                .subscribe(first => this.setState({ first, loading: true }, () => refreshRequests.next()))
+                .pipe(
+                    map(() =>
+                        // If after is set, then it's the endCursor from the previous request.
```

Comment:
Nit: this makes it clearer for me that you're talking about the `after` variable, and that there's no "bug" in the sentence 🙂 

```suggestion
                        // If `after` is set, then it's the endCursor from the previous request.
```